File: packages/tasks-gen/snippets-fixtures/conversational-llm-stream/js/openai/0.together.js
import { OpenAI } from "openai";

const client = new OpenAI({
	baseURL: "https://api.together.xyz/v1",
	apiKey: process.env.TOGETHER_API_TOKEN,
});

const stream = await client.chat.completions.create({
    model: "<together alias for meta-llama/Llama-3.1-8B-Instruct>",
    messages: [
        {
            role: "user",
            content: "What is the capital of France?",
        },
    ],
    stream: true,
});

for await (const chunk of stream) {
    process.stdout.write(chunk.choices[0]?.delta?.content || "");
}

File: packages/tasks-gen/snippets-fixtures/conversational-llm-stream/python/openai/0.together.py
import os
from openai import OpenAI

client = OpenAI(
    base_url="https://router.huggingface.co/v1",
    api_key=os.environ["HF_TOKEN"],
)

stream = client.chat.completions.create(
    model="meta-llama/Llama-3.1-8B-Instruct:together",
    messages=[
        {
            "role": "user",
            "content": "What is the capital of France?"
        }
    ],
    stream=True,
)

for chunk in stream:
    print(chunk.choices[0].delta.content, end="")